app/forms/claimants_detail.rb

```diff
@@ -46,11 +46,11 @@ def disagree_with_employment_dates?
   end
 
   def end_date_is_after_start_date
-    return if employment_end.blank? || employment_start.blank?
+    return if employment_end.blank? || employment_start.blank? || employment_start.is_a?(EtDateType::InvalidDate) || employment_end.is_a?(EtDateType::InvalidDate)
```

Comment:
I don't think this is correct.  If the method is called end_date_is_after_start_date, why are we checking for EtDateType::InvalidDate ?  That has nothing to do with calculating end date is after start date.

If we have an invalid date, it should be caught as early as possible and thrown back to the user as an invald date, then all other code doesnt need to worry about it

Reply:
I added this in the case there is a valid date and an invalid date. Without checking for the EtDateType::InvalidDate type, the validations were error during comparison (line 51).

I thought about adding another early return when one of the start or end dates was EtDateType::InvalidDate, but wasn't sure whether to add here or as a separate validation check? I'm open to suggestions